crates/uv/src/main.rs

```diff
@@ -651,6 +651,10 @@ struct VenvArgs {
     #[clap(default_value = ".venv")]
     name: PathBuf,
 
+    /// The prompt used in shells.
+    #[clap(long, default_value = ".")]
+    prompt: String,
```

Comment:
If I understand this correctly, then we would always default the prompt value to use the current directory name. This is different than the original behavior. What we want is:

1. If the `--prompt` flag isn't present, use the virtual environment name as the prompt value (what we currently do)
2. If the `--prompt` value is ".", use the current directory name
3. Otherwise, use whatever the value is provided by the user

This implementation is missing (1) behavior. This means that we would want to have an `Option<String>` here instead. If it's `None`, then we'd go with (1) otherwise we'd go with either (2) or (3). You will need to change other references accordingly.

```suggestion
    /// The prompt used in shells.
    #[clap(long)]
    prompt: Option<String>,
```

Reply:
> * If the `--prompt` flag isn't present, use the virtual environment name as the prompt value (what we currently do)

There is an important difference between venv/virtualenv and uv venv.

venv/virtualenv was born before `.venv` become convention so they have mandatory ENV_DIR argument. It is natural to use ENV_DIR as the default prompt name.

On the other hand, uv venv uses `.venv` as ENV_DIR by default. But ".venv" is not good for default prompt.

How about changing default prompt by `name`? `prompt = if name == ".venv" { cwd.file_name() } else { name }`

Reply:
> venv/virtualenv was born before `.venv` become convention so they have mandatory ENV_DIR argument. It is natural to use ENV_DIR as the default prompt name.

Good point. Thanks for giving me the context on your decision.

> On the other hand, uv venv uses `.venv` as ENV_DIR by default. But ".venv" is not good for default prompt.

This is why `--prompt` will exist ;)

What happens when a user provided a custom name for the virtual environment directory but the prompt still defaults to the current directory?

```console
$ cargo run --bin uv -- venv hello
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `target/debug/uv venv hello`
Using Python 3.12.1 interpreter at /Users/dhruv/.pyenv/versions/3.12.1/bin/python3.12
Creating virtualenv at: hello

$ source hello/bin/activate

$ bat .venv/pyvenv.cfg | grep prompt
prompt = uv

$ echo $VIRTUAL_ENV_PROMPT 
hello
```

I thought about what you've proposed but I still think we should stick with the original behavior for compatibility reason, not just with virtualenv / venv but with other tools which uses this information from either `VIRTUAL_ENV_PROMPT` or `pyvenv.cfg` file.

Reply:
> What happens when a user provided a custom name for the virtual environment directory but the prompt still defaults to the current directory?

Then, current PR doesn't set prompt. Activation script uses the venv directory name. This is same to venv/virtualenv.

```
$ ../target/debug/uv venv -p 3.11 hello
Using Python 3.11.7 interpreter at /opt/homebrew/opt/python@3.11/bin/python3.11
Creating virtualenv at: hello

$ grep prompt hello/pyvenv.cfg

$ source hello/bin/activate

$ echo $VIRTUAL_ENV_PROMPT
hello
```

Reply:
The only difference in this PR from original virtualenv/venv is:

* if target directory is `.venv` (default), and
* if prompt is not specified,
* uv assumes `--prompt .` is specified.

Reply:
Oh sorry, my example was incorrect. Yours is right.

I thought about this and I think what you proposed is a good default behavior:
* By default, use the current working directory as the prompt value
* If the virtual environment path is provided, then use that as the prompt value
* If `--prompt` is given
	* Value is `.`, use current working directory name
	* Otherwise, use the value provided